### DISCUSSION TITLE
Possible limitation using a local Timestamp Authority (TSA)?

### DISCUSSION THREAD

**Discussion by jsuto** [openssl/openssl#24555] [Q&A] [answered]
Let's say, I have a document that I need to prove its authenticity in court. So I create a hash that I sign with a local Timestamp Authority (TSA), and store the response. Then I show the openssl ts -verify output in the court.

Now, let's say that the opposing party in court accuses me that I have changed the document, signed it again by the local TSA, and no wonder that the verify still gives "OK" result. And since it's a local TSA in my control I could have fixed the time as well.

Is it true that you can claim that "that's impossible" only, if you use a 3rd party TSA service?

**Comment by t8m** [comment, marked as answer]
Yes, the timestamp authority must be a trusted 3rd party service in the same way as certificate authorities are.
